README.md

```diff
@@ -1,555 +1,242 @@
+
+#### Observing text edits
+
+The first step is to observe edits to the text field, using a RAC extension to
+`UITextField` specifically for this purpose:
 
-Signals can also represent timers, other UI events, or anything else that
-changes over time.
-
-Using signals for asynchronous operations makes it possible to build up more
-complex behavior by chaining and transforming those signals. Work can easily be
-triggered after a group of operations completes:
-
-```objc
-// Performs 2 network operations and logs a message to the console when they are
-// both completed.
-//
-// +merge: takes an array of signals and returns a new RACSignal that passes
-// through the values of all of the signals and completes when all of the
-// signals complete.
-//
-// -subscribeCompleted: will execute the block when the signal completes.
-[[RACSignal 
-	merge:@[ [client fetchUserRepos], [client fetchOrgRepos] ]] 
-	subscribeCompleted:^{
-		NSLog(@"They're both done!");
-	}];
+```swift
+let searchStrings = textField.rac_textSignal()
+    |> toSignalProducer()
+    |> map { text in text as! String }
 ```
 
-Signals can be chained to sequentially execute asynchronous operations, instead
-of nesting callbacks with blocks. This is similar to how [futures and promises][]
-are usually used:
-
-```objc
-// Logs in the user, then loads any cached messages, then fetches the remaining
-// messages from the server. After that's all done, logs a message to the
-// console.
-//
-// The hypothetical -logInUser methods returns a signal that completes after
-// logging in.
-//
-// -flattenMap: will execute its block whenever the signal sends a value, and
-// returns a new RACSignal that merges all of the signals returned from the block
-// into a single signal.
-[[[[client 
-	logInUser] 
-	flattenMap:^(User *user) {
-		// Return a signal that loads cached messages for the user.
-		return [client loadCachedMessagesForUser:user];
-	}]
-	flattenMap:^(NSArray *messages) {
-		// Return a signal that fetches any remaining messages.
-		return [client fetchMessagesAfterMessage:messages.lastObject];
-	}]
-	subscribeNext:^(NSArray *newMessages) {
-		NSLog(@"New messages: %@", newMessages);
-	} completed:^{
-		NSLog(@"Fetched all messages.");
-	}];
+This gives us a [signal
+producer](Documentation/FrameworkOverview.md#signal-producers) which sends
+values of type `String`. _(The cast is [currently
+necessary](https://github.com/ReactiveCocoa/ReactiveCocoa/issues/2182) to bridge
+this extension method from Objective-C.)_
+
+#### Making network requests
+
+With each string, we want to execute a network request. Luckily, RAC offers an
+`NSURLSession` extension for doing exactly that:
+
+```swift
+let searchResults = searchStrings
+    |> flatMap(.Latest) { query in
+        let URLRequest = self.searchRequestWithEscapedQuery(query)
+        return NSURLSession.sharedSession().rac_dataWithRequest(URLRequest)
+    }
+    |> map { data, URLResponse in
```

Comment:
I was trying to make the type clear without resorting to (quite noisy) annotations. I can change it if it’s not helpful.